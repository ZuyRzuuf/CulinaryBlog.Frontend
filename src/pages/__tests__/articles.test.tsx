import { render } from '@testing-library/react';
import ArticlesPage, { getServerSideProps } from '../articles';
import mockArticlesData from '../../data_mocks/articles.json';

describe('ArticlesPage', () => {
    it('matches snapshot', () => {
        const props = { articles: mockArticlesData };
        const { asFragment } = render(<ArticlesPage {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('getServerSideProps returns expected data', async () => {
        const { props } = await getServerSideProps(undefined);
        expect(props.articles).toEqual(mockArticlesData);
    });
});
