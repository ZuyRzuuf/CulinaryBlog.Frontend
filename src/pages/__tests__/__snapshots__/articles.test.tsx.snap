// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesPage matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-gray-800 text-white p-4"
  >
    <nav>
      <a
        href="/"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Home
        </span>
      </a>
      <a
        href="/recipes"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Recipes
        </span>
      </a>
      <a
        href="/articles"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Articles
        </span>
      </a>
      <a
        href="/about"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          About
        </span>
      </a>
      <a
        href="/contact"
      >
        <span
          class="mx-2 cursor-pointer"
        >
          Contact
        </span>
      </a>
    </nav>
  </header>
  <main
    class="p-4"
  >
    <section
      class="my-16"
    >
      <h1
        class="text-3xl font-bold mb-8"
      >
        Articles
      </h1>
      <div
        class="mb-8"
      />
      <div>
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4"
        >
          <a
            href="/article/1"
          >
            <span
              class="block bg-white shadow-lg rounded-lg overflow-hidden"
            >
              <img
                alt="Exploring the Flavors of Thai Cuisine"
                class="w-full h-56 object-cover"
                src="/images/articles/1/main.jpeg"
              />
              <div
                class="p-4"
              >
                <h2
                  class="text-xl font-bold mb-2"
                >
                  Exploring the Flavors of Thai Cuisine
                </h2>
                <div
                  class="text-sm text-gray-500"
                >
                  <span>
                    By john_doe
                  </span>
                  <span>
                     • 
                  </span>
                  <span>
                    Posted on 2023-04-10
                  </span>
                </div>
              </div>
            </span>
          </a>
          <a
            href="/article/2"
          >
            <span
              class="block bg-white shadow-lg rounded-lg overflow-hidden"
            >
              <img
                alt="The Art of Italian Pasta"
                class="w-full h-56 object-cover"
                src="/images/articles/2/main.jpeg"
              />
              <div
                class="p-4"
              >
                <h2
                  class="text-xl font-bold mb-2"
                >
                  The Art of Italian Pasta
                </h2>
                <div
                  class="text-sm text-gray-500"
                >
                  <span>
                    By john_doe
                  </span>
                  <span>
                     • 
                  </span>
                  <span>
                    Posted on 2023-04-12
                  </span>
                </div>
              </div>
            </span>
          </a>
        </div>
      </div>
      <div
        class="mt-8"
      />
    </section>
  </main>
  <footer
    class="bg-gray-800 text-white p-4"
  >
    <p>
      © 2023 PanPrzyGarach
    </p>
  </footer>
</DocumentFragment>
`;
